test/uninsured-vesting-v1/uninsured-vesting-v1.test.ts

```diff
@@ -160,50 +161,92 @@ describe("VestingV1", () => {
       });
     });
 
+    describe("transfer project role", () => {
+      it("should only be transferable by project wallet", async () => {
+        expect(await vesting.methods.projectWallet().call()).to.be.eq(projectWallet);
+        await vesting.methods.transferProjectRole(differentProjectWallet).send({ from: projectWallet });
+        expect(await vesting.methods.projectWallet().call()).to.be.eq(differentProjectWallet);
+      });
+
+      it("should not be updatable to zero address", async () => {
+        await expectRevert(
+          () => vesting.methods.transferProjectRole(zeroAddress).send({ from: projectWallet }),
+          "ProjectRole: new project wallet is the zero address"
+        );
+      });
+
+      it("project role permissions should be available to new project wallet address after role transfer", async () => {
+        await vesting.methods.setAmount(user1, await projectToken.amount(TOKENS_PER_USER)).send({ from: projectWallet });
+        await vesting.methods.activate(await getDefaultStartTime()).send({ from: projectWallet });
+        await vesting.methods.transferProjectRole(differentProjectWallet).send({ from: projectWallet });
+        await advanceDays(10);
+        await vesting.methods.claim(user1).send({ from: differentProjectWallet });
+      });
+
+      it("old wallet should not have project role permissions after ownership transfer", async () => {
+        await vesting.methods.setAmount(user1, await projectToken.amount(TOKENS_PER_USER)).send({ from: projectWallet });
+        await vesting.methods.transferProjectRole(differentProjectWallet).send({ from: projectWallet });
+
+        await expectRevert(
+          async () => await vesting.methods.setAmount(user1, await projectToken.amount(TOKENS_PER_USER)).send({ from: projectWallet }),
+          PROJECT_ROLE_REVERT_MSG
+        );
+
+        await expectRevert(async () => await vesting.methods.claim(user1).send({ from: projectWallet }), Error.OnlyProjectOrSender);
+      });
+
+      it("old wallet address should not be able to call transfer role again after initial transfer", async () => {
+        await vesting.methods.setAmount(user1, await projectToken.amount(TOKENS_PER_USER)).send({ from: projectWallet });
+        await vesting.methods.transferProjectRole(differentProjectWallet).send({ from: projectWallet });
+
+        await expectRevert(async () => await vesting.methods.activate(await getDefaultStartTime()).send({ from: projectWallet }), PROJECT_ROLE_REVERT_MSG);
+      });
+    });
+
```

Comment:
maybe this block (I believe a similar one exists for Insured) should be in a separate file, testing `ProjectSol` only, and we can settle for testing it only once.

(i.e. if ProjectSol tests fine, and VestingV1 is ProjectSol, then it's covered)

Reply:
Sounds OK to me, but will that influence our test coverage results?